drivers/sensor/fdc2x1x/fdc2x1x.c

```diff
@@ -0,0 +1,1010 @@
+		return -ENODEV;
+	}
+
+	if (fdc2x1x_reg_read(dev, FDC2X1X_MANUFACTURER_ID, &man_id) < 0) {
+		LOG_ERR("Failed to read manufacturer id");
+		return -EIO;
+	}
+
+	if (man_id != FDC2X1X_MANUFACTURER_ID_VAL) {
+		LOG_ERR("Wrong manufaturer id");
+		return -ENODEV;
+	}
+
+	return 0;
+}
+
+/**
+ * Initialize the SD-Pin.
+ * @param dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+static int fdc2x1x_init_sd_pin(const struct device *dev)
+{
+	struct fdc2x1x_data *data = dev->data;
+	const struct fdc2x1x_config *cfg = dev->config;
+
+	data->gpio = device_get_binding(cfg->sd_name);
+	if (!data->gpio) {
+		LOG_ERR("Failed to get pointer to %s device!", cfg->sd_name);
+		return -EINVAL;
+	}
+
+	gpio_pin_configure(data->gpio, cfg->sd_pin,
+			   GPIO_OUTPUT_INACTIVE | cfg->sd_flags);
+
+	gpio_pin_set(data->gpio, cfg->sd_pin, 0);
+
+	return 0;
+}
+
+/**
+ * Initialization of the device.
+ * @param dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+static int fdc2x1x_init(const struct device *dev)
+{
+	struct fdc2x1x_data *data = dev->data;
+	const struct fdc2x1x_config *cfg = dev->config;
+	uint8_t ch_supported;
+
+	if (cfg->fdc2x14) {
+		ch_supported = 4;
+	} else {
+		ch_supported = 2;
+	}
+
+	if ((cfg->num_channels <= 0)) {
+		LOG_ERR("No channel nodes found");
+		return -EINVAL;
+	} else if (cfg->num_channels > ch_supported) {
+		LOG_ERR("Amount of channels not supported by this device");
+		return -EINVAL;
+	}
+
+	if (cfg->sd_name) {
+		if (fdc2x1x_init_sd_pin(dev) < 0) {
+			return -EINVAL;
+		}
+	}
+
+	data->bus = device_get_binding(cfg->i2c_name);
+	if (data->bus  == NULL) {
+		LOG_ERR("Failed to get pointer to %s device!", cfg->i2c_name);
+		return -EINVAL;
+	}
+
+	k_sleep(K_MSEC(100));
+
+	if (fdc2x1x_probe(dev) < 0) {
+		return -ENODEV;
+	}
+
+	if (fdc2x1x_reset(dev) < 0) {
+		return -EIO;
+	}
+
+	if (fdc2x1x_init_config(dev) < 0) {
+		return -EIO;
+	}
+
+	if (fdc2x1x_set_op_mode(dev, FDC2X1X_ACTIVE_MODE) < 0) {
+		return -EIO;
+	}
+#ifdef CONFIG_PM_DEVICE
+	data->pm_state = FDC2X1X_ACTIVE_MODE;
+#endif
+
+#ifdef CONFIG_FDC2X1X_TRIGGER
+	if (fdc2x1x_init_interrupt(dev) < 0) {
+		LOG_ERR("Failed to initialize interrupt!");
+		return -EIO;
+	}
+#endif
+
+	return 0;
+}
+
+#define FDC2X1X_SD_PROPS(n)			     \
+	.sd_name = DT_INST_GPIO_LABEL(n, sd_gpios),  \
+	.sd_pin = DT_INST_GPIO_PIN(n, sd_gpios),     \
+	.sd_flags = DT_INST_GPIO_FLAGS(n, sd_gpios), \
+
+#define FDC2X1X_SD(n)					\
+	COND_CODE_1(DT_INST_NODE_HAS_PROP(n, sd_gpios),	\
+		    (FDC2X1X_SD_PROPS(n)),		\
+		    ())
+
+#define FDC2X1X_INTB_PROPS(n)				 \
+	.intb_name = DT_INST_GPIO_LABEL(n, intb_gpios),	 \
+	.intb_pin = DT_INST_GPIO_PIN(n, intb_gpios),	 \
+	.intb_flags = DT_INST_GPIO_FLAGS(n, intb_gpios), \
+
+#define FDC2X1X_INTB(n)			     \
+	COND_CODE_1(CONFIG_FDC2X1X_TRIGGER,  \
+		    (FDC2X1X_INTB_PROPS(n)), \
+		    ())
+
+#define FDC2X1X_CH_CFG_INIT(m)				  \
```

Comment:
I think `ch` would be a better variable name than `m`. Here, `m` after `n` suggests that `m` is also an instance number, but it's actually a node identifier.

Reply:
changed the variable name to `ch`